script/update-content-versions-and-check-for-archived-slugs.ts

```diff
@@ -0,0 +1,105 @@
+import fs from 'fs';
+import fetch from 'node-fetch';
+import path from 'path';
+import { argv } from 'yargs';
+import { LinkedArchiveTreeNode } from '../src/app/content/types';
+import { flattenArchiveTree } from '../src/app/content/utils';
+import archivedSlugs from '../src/archived-slugs.json';
+import { ARCHIVE_URL, REACT_APP_ARCHIVE_URL, REACT_APP_OS_WEB_API_URL } from '../src/config';
+import books from '../src/config.books.json';
+import createArchiveLoader from '../src/gateways/createArchiveLoader';
+import createOSWebLoader from '../src/gateways/createOSWebLoader';
+
+interface ArchivedSlug {
+  pathname: string;
+  bookId: string;
+  pageId: string;
+}
+
+type ArchivedSlugs = ArchivedSlug[];
+
+const { book, newVersion } = argv as any as {
+  book: string
+  newVersion: string | number;
+};
+
+(global as any).fetch = fetch;
+
+const booksPath = path.resolve(__dirname, '../src/config.books.json');
+const archivedSlugsPath = path.resolve(__dirname, '../src/archived-slugs.json');
+
+const archiveLoader = createArchiveLoader(`${ARCHIVE_URL}${REACT_APP_ARCHIVE_URL}`);
+const osWebLoader = createOSWebLoader(`${ARCHIVE_URL}${REACT_APP_OS_WEB_API_URL}`);
+
+async function processBooks() {
+  const booksData = Object.entries(books);
+  const bookToUpdateIndex = booksData.findIndex(([id]) => id === book);
+  const bookToUpdate = booksData[bookToUpdateIndex] as [string, { defaultVersion: string }] | undefined;
+
+  if (!bookToUpdate) {
+    console.error(`book ${book} not found`); // tslint:disable-line:no-console
+    return;
+  }
+
+  const [bookId, { defaultVersion }] = bookToUpdate;
+
+  if (defaultVersion === newVersion.toString()) {
+    console.log(`$${bookId} alredy at desired version.`); // tslint:disable-line:no-console
+    return;
+  }
+
+  const { title, tree: currentTree } = await archiveLoader.book(bookId, defaultVersion).load()
+    .catch((error) => {
+      // tslint:disable-next-line: no-console
+      console.log(`error while loading book ${bookId} with defaultVersion ${defaultVersion}`);
+      throw error;
+    });
+  const { tree: newTree } = await archiveLoader.book(bookId, newVersion.toString()).load()
+    .catch((error) => {
+      console.log(`error while loading book ${bookId} with newVersion ${newVersion}`); // tslint:disable-line:no-console
+      throw error;
+    });
+  const bookSlug = await osWebLoader.getBookSlugFromId(bookId)
+    .catch((error) => {
+      console.log(`error while loading slug for bookId ${bookId}`); // tslint:disable-line:no-console
+      throw error;
+    });
+
+  const flatCurrentTree = flattenArchiveTree(currentTree);
+  const flatNewTree = flattenArchiveTree(newTree);
+  const missingPages = [...archivedSlugs] as ArchivedSlugs;
```

Comment:
why copy this array and not just use the existing variable? if you're just trying to assign a type you might want to instead create a definition file for the redirects file like the books.json has